### PR TITLE
Pass inline check traits to Statistic and CheckPF2e

### LINKED ISSUE #3047: Feature Request: show the traits used in DC Rollable Buttons in the roll outcome
It would be nice if you could see the traits used in the dc rollable buttons e.g. @Check[type:diplomacy|dc:20|traits:auditory,concentrate,emotion,exploration,linguistic,mental|showDC:gm|name:Coerce] in the outcome of the roll as well (like the skill action macros):
skillissue

### PR DESCRIPTION
**Pull request by In3luki**
`@Check[type:diplomacy|dc:20|traits:auditory,concentrate,emotion,exploration,linguistic,mental|showDC:gm|name:Coerce] `
image

Closes #3047

### CODE REVIEW

File: src/scripts/ui/inline-roll-links.ts [hunk 1]
```diff
@@ -78,97 +76,89 @@ export const InlineRollLinks = {
...
                 return;
             }
-
+            const creatureActors = actors.filter((actor): actor is CreaturePF2e => actor.isOfType("creature"));
+            const parsedTraits = pf2Traits
+                ? pf2Traits
+                      .split(",")
+                      .map((trait) => trait.trim())
+                      .filter((trait) => !!trait)
+                : undefined;
```
Comment (stwlam) [inline src/scripts/ui/inline-roll-links.ts:85]:
```suggestion
            const parsedTraits = pf2Traits
                ?.split(",")
                .map((trait) => trait.trim())
                .filter((trait) => !!trait);
```

File: src/scripts/ui/inline-roll-links.ts [hunk 2]
```diff
@@ -78,97 +76,89 @@ export const InlineRollLinks = {
...
-                                        .map((trait) => trait.trim())
-                                        .filter((trait) => !!trait);
-                                    options.push(...traits);
-                                }
-                                flatCheck.check.roll({ ...eventRollParams, extraRollOptions: options, dc });
-                            } else {
-                                console.warn(`PF2e System | Skip rolling flat check for '${actor}'`);
-                            }
+                        } else {
+                            console.warn(`PF2e System | Skip rolling flat check for '${actor}'`);
```
Comment (stwlam) [inline src/scripts/ui/inline-roll-links.ts:130]:
```suggestion
                            console.warn(`PF2e System | Skip rolling flat check for "${actor}"`);
```

File: src/module/system/statistic/index.ts [hunk 3]
```diff
@@ -342,6 +349,11 @@ class StatisticCheck {
             }
         }
 
+        // Add additional traits
+        const traitObjects = args.traits
+            ? args.traits.map((trait) => traitSlugToObject(trait, CONFIG.PF2E.actionTraits))
+            : [];
```
Comment (stwlam) [inline src/module/system/statistic/index.ts:355]:
```suggestion
                const traitObjects = args.traits?.map((trait) => traitSlugToObject(trait, CONFIG.PF2E.actionTraits)) ?? [];
```

**Comment by stwlam** [comment]
Actually, I don't think we can merge this. The "traits" from inline thingies aren't actually traits but roll options, some of which happen to correspond with traits. They're not traits for the check.

**Comment by In3luki** [comment]
True, there would have to be separate option for real traits but then the current `traits` would have to be renamed to `rollOptions` which would require a migration. Probably not worth it for a minimal gain.

I'll open a new PR with the cleaned up `inline-roll-links.ts` later.

**Pull request closed without merging**